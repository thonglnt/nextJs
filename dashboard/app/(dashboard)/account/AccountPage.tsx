'use client'

import {
  Box,
  Button,
  Card,
  Container,
  Divider,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Typography,
  useTheme,
} from '@mui/material'
import { useState } from 'react'
import ArrowForwardIcon from '@mui/icons-material/ArrowForward'

// Components
import Avatar from '@components/Avatar'
import CardActions from '@components/CardActions'
import CardHeader from '@components/CardHeader'
import Input from '@components/Input'
import Logo from '@components/Logo'
import Switch from '@components/Switch'

// Constants
import { MOCK_INVOICE, MOCK_LOGIN_HISTORY, ROUTE } from '@constants/index'

const AccountPage = () => {
  const theme = useTheme()
  const [toggle, setToggle] = useState('general')

  const renderAccountContent = () => {
    if (toggle === 'general') {
      return (
        <Box marginTop="32px">
          {/* Basic details */}
          <Card
            sx={{
              padding: '32px 24px',
              display: 'flex',
              justifyContent: 'space-between',
            }}
          >
            <Typography
              fontSize="18px"
              fontWeight={700}
              lineHeight="25px"
              width="30%"
            >
              Basic details
            </Typography>
            <Box width="69%" display="flex" flexDirection="column" gap="20px">
              <Box display="flex" gap="20px">
                <Avatar
                  type="normal"
                  sx={{
                    width: '56px',
                    height: '56px',
                  }}
                />
                <Button
                  variant="text"
                  color="success"
                  sx={{
                    textTransform: 'none',
                  }}
                >
                  change
                </Button>
              </Box>
              <Box display="flex">
                <Input
                  fullWidth
                  label="Full name"
                  name="fullName"
                  defaultValue="Shen Zen"
                />
                <Button
                  variant="text"
                  sx={{
                    textTransform: 'none',
                  }}
                  disabled
                >
                  Save
                </Button>
              </Box>
              <Box display="flex">
                <Input
                  fullWidth
                  label="Email address"
                  name="email"
                  defaultValue="shen.zen@acme.com"
                  disabled
                />
                <Button
                  variant="text"
                  color="secondary"
                  sx={{
                    textTransform: 'none',
                  }}
                >
                  Edit
                </Button>
              </Box>
            </Box>
          </Card>

          {/* Public profile */}
          <Card
            sx={{
              marginTop: '32px',
              padding: '32px 24px',
              display: 'flex',
              justifyContent: 'space-between',
            }}
          >
            <Typography
              fontSize="18px"
              fontWeight={700}
              lineHeight="25px"
              width="30%"
            >
              Public profile
            </Typography>
            <Box width="69%" display="flex" flexDirection="column" gap="24px">
              <Box display="flex">
                <CardHeader
                  title="Make Contact info Public"
                  desc="Means that anyone viewing yo your profile will be able to see your contacts details"
                />
                <Switch />
              </Box>
              <Divider />
              <Box display="flex">
                <CardHeader
                  title="Available to hire"
                  desc="Toggling this will let your teammates know that you are available for acquiring new projects"
                />
                <Switch defaultChecked />
              </Box>
            </Box>
          </Card>

          {/* Deleted account */}
          <Card
            sx={{
              marginTop: '32px',
              padding: '32px 24px',
              display: 'flex',
              justifyContent: 'space-between',
            }}
          >
            <Typography
              fontSize="18px"
              fontWeight={700}
              lineHeight="25px"
              width="30%"
            >
              Deleted account
            </Typography>
            <Box width="69%" display="flex" flexDirection="column" gap="20px">
              <Typography fontSize="16px" fontWeight={700} lineHeight="28px">
                Delete your account and all of your source data. This is
                irreversible.
              </Typography>
              <Button variant="outlined" color="error">
                Deleted account
              </Button>
            </Box>
          </Card>
        </Box>
      )
    }

    if (toggle === 'billing') {
      return (
        <Box marginTop="32px">
          <Card
            sx={{
              padding: '32px 24px',
            }}
          >
            <CardHeader
              title="Change plan"
              desc="You can upgrade and downgrade whenever you want."
            />

            <Box display="flex" justifyContent="space-between" gap="30px">
              <Card
                sx={{
                  padding: '32px 24px',
                  width: '100%',
                }}
              >
                <Logo />
                <Box
                  display="flex"
                  alignItems="center"
                  color={theme.palette.grey[600]}
                  marginTop="37px"
                  gap="4px"
                >
                  <Typography
                    fontSize="24px"
                    fontWeight={800}
                    lineHeight="33px"
                    color="black"
                  >
                    $0
                  </Typography>{' '}
                  /mo
                </Box>
                <Box display="flex" justifyContent="space-between">
                  <Typography fontSize="12px" lineHeight="30px">
                    STARTUP
                  </Typography>
                  <CardActions
                    title="Using now"
                    href={ROUTE.DEFAULT}
                    color={theme.palette.success.light}
                  />
                </Box>
              </Card>
              <Card
                sx={{
                  padding: '32px 24px',
                  width: '100%',
                }}
              >
                <Logo />
                <Box
                  display="flex"
                  alignItems="center"
                  color={theme.palette.grey[600]}
                  marginTop="37px"
                  gap="4px"
                >
                  <Typography
                    fontSize="24px"
                    fontWeight={800}
                    lineHeight="33px"
                    color="black"
                  >
                    $4.99
                  </Typography>{' '}
                  /mo
                </Box>
                <Typography fontSize="12px" lineHeight="30px">
                  STANDARD
                </Typography>
              </Card>
              <Card
                sx={{
                  padding: '32px 24px',
                  width: '100%',
                }}
              >
                <Logo />
                <Box
                  display="flex"
                  alignItems="center"
                  color={theme.palette.grey[600]}
                  marginTop="37px"
                  gap="4px"
                >
                  <Typography
                    fontSize="24px"
                    fontWeight={800}
                    lineHeight="33px"
                    color="black"
                  >
                    $29.99
                  </Typography>{' '}
                  /mo
                </Box>
                <Typography fontSize="12px" lineHeight="30px">
                  BUSINESS
                </Typography>
              </Card>
            </Box>
            <Divider
              sx={{
                margin: '24px 0',
              }}
            />
            <CardHeader
              title="Invoice history"
              desc="You can view and download all your previous invoices here. If you've just made a payment, it may take a few hours for it to appear in the table below."
            />
            <Table>
              <TableHead>
                <TableRow>
                  <TableCell>DATE</TableCell>
                  <TableCell>TOTAL (INCL. TAX)</TableCell>
                  <TableCell />
                </TableRow>
              </TableHead>
              <TableBody>
                {MOCK_INVOICE.map((item) => (
                  <TableRow key={item.id}>
                    <TableCell>{item.date}</TableCell>
                    <TableCell>{`$${item.total}`}</TableCell>
                    <TableCell align="right">
                      <CardActions title="View invoice" href={ROUTE.DEFAULT} />
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </Card>
        </Box>
      )
    }

    if (toggle === 'security') {
      return (
        <Box marginTop="32px">
          <Card
            sx={{
              padding: '32px 24px',
              display: 'flex',
              justifyContent: 'space-between',
            }}
          >
            <Typography
              fontSize="18px"
              fontWeight={700}
              lineHeight="25px"
              width="30%"
            >
              Change password
            </Typography>
            <Box width="69%" display="flex" gap="20px">
              <Input
                fullWidth
                label="Password"
                name="password"
                defaultValue="12312312312"
                type="password"
                disabled
              />
              <Switch defaultChecked />
            </Box>
          </Card>
          <Card
            sx={{
              marginTop: '32px',
              padding: '32px 24px',
            }}
          >
            <Typography fontSize="18px" fontWeight={700} lineHeight="25px">
              Multi Factor Authentication
            </Typography>
            <Box marginTop="24px" display="flex" gap="32px">
              <Card
                sx={{
                  padding: '36px 24px',
                  width: '100%',
                  display: 'flex',
                  flexDirection: 'column',
                  gap: '16px',
                }}
              >
                <Typography
                  color={theme.palette.error.light}
                  fontSize="14px"
                  lineHeight="22px"
                >
                  Off
                </Typography>
                <CardHeader
                  title="Authenticator App"
                  desc="Use an authenticator app to generate one time security codes."
                />

                <Button
                  endIcon={<ArrowForwardIcon />}
                  variant="outlined"
                  color="secondary"
                  sx={{
                    maxWidth: '110px',
                  }}
                >
                  Set Up
                </Button>
              </Card>
              <Card
                sx={{
                  padding: '36px 24px',
                  width: '100%',
                  display: 'flex',
                  flexDirection: 'column',
                  gap: '16px',
                }}
              >
                <Typography
                  color={theme.palette.error.light}
                  fontSize="14px"
                  lineHeight="22px"
                >
                  Off
                </Typography>
                <CardHeader
                  title="Text Message"
                  desc="Use your mobile to receive security code via SMS."
                />
                <Button
                  endIcon={<ArrowForwardIcon />}
                  variant="outlined"
                  color="secondary"
                  sx={{
                    maxWidth: '110px',
                  }}
                >
                  Set Up
                </Button>
              </Card>
            </Box>
          </Card>
          <Card
            sx={{
              marginTop: '32px',
              padding: '32px 24px',
            }}
          >
            <CardHeader
              title="Login history"
              desc="Your recent login activity:"
            />
            <Table>
              <TableHead>
                <TableRow>
                  <TableCell>LOGIN TYPES</TableCell>
                  <TableCell>IP ADDRESS</TableCell>
                  <TableCell>CLIENTS</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {MOCK_LOGIN_HISTORY.map((item) => (
                  <TableRow key={item.id}>
                    <TableCell>
                      <CardHeader title="Credentials login" desc={item.type} />
                    </TableCell>
                    <TableCell>{item.ip}</TableCell>
                    <TableCell>{item.client}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </Card>
        </Box>
      )
    }
    return <Box>No content</Box>
  }

  return (
    <Container
      maxWidth="lg"
      sx={{
        padding: '64px 0',
      }}
    >
      <Box>
        <Typography
          variant="h2"
          fontSize="32px"
          lineHeight="38px"
          fontWeight={800}
        >
          Account
        </Typography>
      </Box>
      <Box marginTop="44px">
        <Box display="flex" gap="32px">
          <Button
            onClick={() => setToggle('general')}
            sx={{
              textTransform: 'capitalize',
            }}
          >
            <Typography
              fontSize="18px"
              lineHeight="24px"
              fontWeight={500}
              color={theme.palette.secondary.main}
            >
              General
            </Typography>
          </Button>
          <Button
            onClick={() => setToggle('billing')}
            sx={{
              textTransform: 'capitalize',
            }}
          >
            <Typography
              fontSize="18px"
              lineHeight="24px"
              fontWeight={500}
              color={theme.palette.secondary.main}
              sx={{
                cursor: 'pointer',
              }}
            >
              Billing
            </Typography>
          </Button>
          <Button
            onClick={() => setToggle('security')}
            sx={{
              textTransform: 'capitalize',
            }}
          >
            <Typography
              fontSize="18px"
              lineHeight="24px"
              fontWeight={500}
              color={theme.palette.secondary.main}
              sx={{
                cursor: 'pointer',
              }}
            >
              Security
            </Typography>
          </Button>
        </Box>
        <Divider
          sx={{
            marginTop: '12px',
          }}
        />
        {renderAccountContent()}
      </Box>
    </Container>
  )
}

export default AccountPage
