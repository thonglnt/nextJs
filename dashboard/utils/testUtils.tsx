import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
