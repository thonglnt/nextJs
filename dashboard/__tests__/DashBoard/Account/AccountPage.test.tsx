import React from 'react'

// Utils
import { fireEvent, render, screen } from '@utils/testUtils'

// App
import AccountPage from 'app/(dashboard)/account/AccountPage'

describe('Account Page', () => {
  it('should render Account Page Component', () => {
    const { container } = render(<AccountPage />)

    const button = screen.getByRole('button', {
      name: /general/i,
    })

    fireEvent.click(button)
    expect(container).toMatchSnapshot()
  })

  it('should render Billing Account Page Component', () => {
    const { container } = render(<AccountPage />)

    const button = screen.getByRole('button', {
      name: /billing/i,
    })

    fireEvent.click(button)

    expect(container).toBeInTheDocument()
  })

  it('should render Security Account Page Component', () => {
    const { container } = render(<AccountPage />)

    const button = screen.getByRole('button', {
      name: /security/i,
    })

    fireEvent.click(button)

    expect(container).toBeInTheDocument()
  })
})
