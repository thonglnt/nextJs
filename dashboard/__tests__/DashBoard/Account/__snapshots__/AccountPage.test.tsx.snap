// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Page should render Account Page Component 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1qmdd7f-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-56movi-MuiTypography-root"
      >
        Account
      </h2>
    </div>
    <div
      class="MuiBox-root css-t1c47c"
    >
      <div
        class="MuiBox-root css-13t3iqc"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1poohah-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1hejyx9-MuiTypography-root"
          >
            General
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1poohah-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-2pgac4-MuiTypography-root"
          >
            Billing
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1poohah-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-2pgac4-MuiTypography-root"
          >
            Security
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-v9rab2-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-h5fkc8"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-eyi1og-MuiPaper-root-MuiCard-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-nh3o6o-MuiTypography-root"
          >
            Basic details
          </p>
          <div
            class="MuiBox-root css-1f412ck"
          >
            <div
              class="MuiBox-root css-kl5uk3"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-8h65ya-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeMedium MuiButton-textSizeMedium css-jwdgh3-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                change
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root css-k008qs"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-xne6jv-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1agiy4x-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="outlined-required"
                  id="outlined-required-label"
                >
                  Full name
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl css-1cswynb-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="outlined-required"
                    name="fullName"
                    type="text"
                    value="Shen Zen"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        Full name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kwi5j1-MuiButtonBase-root-MuiButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                Save
              </button>
            </div>
            <div
              class="MuiBox-root css-k008qs"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-xne6jv-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1agiy4x-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="outlined-required"
                  id="outlined-required-label"
                >
                  Email address
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-1cswynb-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="false"
                    class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled=""
                    id="outlined-required"
                    name="email"
                    type="text"
                    value="shen.zen@acme.com"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        Email address
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-pw23t8-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Edit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qwdabd-MuiPaper-root-MuiCard-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-nh3o6o-MuiTypography-root"
          >
            Public profile
          </p>
          <div
            class="MuiBox-root css-1wa7et"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <div
                class="MuiBox-root css-7pf6at"
              >
                <div
                  class="MuiBox-root css-0"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                  >
                    Make Contact info Public
                  </h5>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                  >
                    Means that anyone viewing yo your profile will be able to see your contacts details
                  </p>
                </div>
              </div>
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
              >
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary css-12f33q6-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </label>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="MuiBox-root css-k008qs"
            >
              <div
                class="MuiBox-root css-7pf6at"
              >
                <div
                  class="MuiBox-root css-0"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                  >
                    Available to hire
                  </h5>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                  >
                    Toggling this will let your teammates know that you are available for acquiring new projects
                  </p>
                </div>
              </div>
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
              >
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked css-12f33q6-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1qwdabd-MuiPaper-root-MuiCard-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-nh3o6o-MuiTypography-root"
          >
            Deleted account
          </p>
          <div
            class="MuiBox-root css-1f412ck"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-oj13xm-MuiTypography-root"
            >
              Delete your account and all of your source data. This is irreversible.
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Deleted account
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
